### ISSUE TITLE
Cannot use trans in panel config after upgrade

### ISSUE THREAD

**Issue by datlechin** [laravel-filament/filament#7604]
### Package

filament/filament

### Package Version

v3.0.14

### Laravel Version

v10.18.0

### PHP Version

PHP 8.2

### Problem description

Cannot use translation in panel config after upgrade from `v3.0.8` to `v3.0.14`

```
Target class [translator] does not exist.
```

```php
            ->userMenuItems([
                MenuItem::make()
                    ->label(__('My Profile')) // error right here
                    ->url(fn () => Profile::getUrl())
                    ->icon('heroicon-o-user-circle'),
            ]);
```

### Expected behavior

works

### Steps to reproduce

Use `trans()` or `__()` in admin panel service

### Reproduction repository

https://github.com/datlechin/filament

**Comment by nielspeen** [comment]
As you came from 3.0.8, I can confirm this starts with 3.0.14. Downgrading to 3.0.13 fixes it.

**Comment by howdu** [comment]
I had a similar issue- solved by using a closure `->label(() => __('My Profile'))`

**Comment by datlechin** [comment]
@howdu how about this
```php
            ->navigationGroups([
                __('Courses'),
                __('Blog'),
                __('Forum'),
                __('Administration'),
            ])
```

**Comment by howdu** [comment]
Use the full class
```php
->navigationGroups([
    NavigationGroup::make()
        ->label(fn () => __('Example'))
        ->collapsible(),
    NavigationGroup::make()
        ->label(fn () => __('Example'))
        ->collapsible(),
])
```

**Comment by datlechin** [comment]
It works btw I don't think this is a good way to solve this issue

**Comment by danharrin** [comment]
Correct me if I'm wrong. In 3.0.13, using `__()` does not actually translate anything, if you set the locale in middleware. It just uses your default locale. You have to actually wrap it in a closure to be able to translate.

**Comment by datlechin** [comment]
ok, you're right. before v3.0.13 translate takes no effect without closure function.

**Comment by danharrin** [comment]
So this is probably a good error lol. It was kinda a coincidence that it worked before. There is no way to access the locale from a service provider, as it is run before middleware.

**Comment by lloricode** [comment]
> Correct me if I'm wrong. In 3.0.13, using `__()` does not actually translate anything, if you set the locale in middleware. It just uses your default locale. You have to actually wrap it in a closure to be able to translate.

Hi!, same issue as mine, using filament `v3.0.15`

It is working on this:

```php
            ->navigationItems([
                NavigationItem::make('API Documentation')
                    ->group(fn () => trans('Documentation'))
```

But this is not working

```php
            ->navigationGroups([
                fn () => trans('Documentation'),
            ])
```

![Screenshot_2023-08-10_08-06-57](https://github.com/filamentphp/filament/assets/8251344/b3384723-61aa-4597-a5f8-96f95bd16c45)

is there any wrong on my setup? hope can fix this

**Comment by lloricode** [comment]
> > Correct me if I'm wrong. In 3.0.13, using `__()` does not actually translate anything, if you set the locale in middleware. It just uses your default locale. You have to actually wrap it in a closure to be able to translate.
> 
> Hi!, same issue as mine, using filament `v3.0.15`
> 
> It is working on this:
> 
> ```
>             ->navigationItems([
>                 NavigationItem::make('API Documentation')
>                     ->group(fn () => trans('Documentation'))
> ```
> 
> But this is not working
> 
> ```
>             ->navigationGroups([
>                 fn () => trans('Documentation'),
>             ])
> ```
> 
> ![Screenshot_2023-08-10_08-06-57](https://user-images.githubusercontent.com/8251344/259579548-b3384723-61aa-4597-a5f8-96f95bd16c45.png)
> 
> is there any wrong on my setup? hope can fix this

My bad, there is a way

```php

            ->navigationGroups([
                NavigationGroup::make()
                    ->label( fn () => trans('Documentation')),
```

**Comment by twarkie** [comment]
I'm forcing my filament language to english with this code in `AdminPanelProvider.php`:

```app()->setLocale('en');```

...which also results in `Target class [translator] does not exist.` after upgrade from 3.0.12 to 3.0.16.

**Comment by danharrin** [comment]
Should be fixed by #7661

**Comment by alphyon** [comment]
I have same issue using v3.0.36
